python/simulation.py

```diff
@@ -33,9 +33,9 @@ def get_num_args(func):
 def py_v3_to_vec(dims, v3):
     if dims == 1:
         if v3.x == v3.y == v3.z:
-            return mp.vec(dims, v3.x)
+            return mp.vec_from_dim(0, v3.x)
         else:
-            return mp.vec(dims)
+            return mp.vec(v3.z)
```

Comment:
What's the difference between `vec_from_dim(0, x)` and `vec(x)`?     And what's the reason for the `v3.x == v3.y == v3.z` check here?

Reply:
I'm not sure what I was thinking WRT the equality statement. I removed it.

`vec_from_dim` is a renaming of `vec(ndim, double)`.  I did this because python can't distinguish between an enum and an int.  If I create a `vec` with `mp.vec(0, 1)`, intending a 2D vec (even though I didn't explicitly pass doubles), SWIG will call `vec(ndim, double)`, which creates a 1D `vec`. Having `vec_from_dim` eliminates the possibility of that mistake.